server/src/main/java/thesistrack/ls1/controller/ThesisController.java

```diff
@@ -0,0 +1,132 @@
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @GetMapping("/{thesisId}/proposal")
+    public ResponseEntity<Resource> getProposalFile(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PutMapping("/{thesisId}/proposal")
+    public ResponseEntity<ThesisDto> uploadProposal(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/proposal/accept")
+    public ResponseEntity<ThesisDto> acceptProposal(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/proposal/reject")
+    public ResponseEntity<ThesisDto> rejectProposal(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/presentations")
+    public ResponseEntity<ThesisDto> createPresentation(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @DeleteMapping("/{thesisId}/presentations/{presentationId}")
+    public ResponseEntity<ThesisDto> deletePresentation(@PathVariable UUID thesisId, @PathVariable UUID presentationId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @GetMapping("/{thesisId}/comments")
+    public ResponseEntity<Page<ThesisDto>> getComments(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/comments")
+    public ResponseEntity<ThesisDto> createComment(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @GetMapping("/{thesisId}/comments/{fileId}")
+    public ResponseEntity<ThesisDto> getCommentFile(@PathVariable UUID thesisId, @PathVariable String fileId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/assessment")
+    public ResponseEntity<ThesisDto> createAssessment(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
+
+    @PostMapping("/{thesisId}/grade")
+    public ResponseEntity<ThesisDto> addGrade(@PathVariable UUID thesisId) {
+        throw new ResponseStatusException(HttpStatus.NOT_IMPLEMENTED, "This feature is not implemented yet");
+    }
```

Comment:
Right now the idea is that they will be part of the Thesis Entity or ThesisDto Object and not a seperate object

Reply:
If you offer the functionality to update/create them separately (=seperate objects), I think you should stay consistent and also offer a GETter for them. Once an API object, always an API object. However, this might introduce a significant (network) overhead when updating the ThesisEntity every time.. Why don't you want to have separate SETter for this?